spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigServerConfigDataLoader.java

```diff
@@ -161,8 +162,9 @@ else if (ALL_OPTIONS.size() > 2) {
 							// https://github.com/spring-cloud/spring-cloud-config/issues/1874
 							for (String profile : resource.getAcceptedProfiles()) {
 								// TODO: switch to match
-								if (propertySourceName.contains("-" + profile + ".")) {
-									// TODO: switch to Options.with() when implemented
+								if (propertySourceName.contains("-" + profile + ".") || propertySourceName
+										.matches("^" + PREFIX + "vault:.*" + "," + profile + "$")) {
```

Comment:
Can you add a comment about why we need this?  It will help with understanding he code

Reply:
I'm very hesitant to add anything repository specific. We should find a more generic way

Reply:
The safest way would be to add an additional argument _profile_ to the property source at the server side and match that on the client against the active profiles. But this is a bigger refactoring. I don't know if you like to make such big changes in a minor or patch release. 

Another option would be to simply check if the propert source name contains the profile name. But i do not think this is a good idea, as there can be false positives.

Reply:
> 

extracted the condition to a private method and add documentation.

Reply:
Why can't we just look for the pattern `,profile_`?

Reply:
We can check for `,profile` (without `_`).
The code can look like this:

```
if(propertySourceName.contains(".*[-,]" + profile + ".*")){
...
}
```

Reply:
Let's go with that as it's similar to the logic already there